tests/test_db_mysql.py

```
@@ -0,0 +1,148 @@
+
+
+class TestFetchAll:
+    """Tests for fetch_all function"""
+
+    def test_fetch_all_returns_tuple(self, mock_db_connection):
+        """Test that fetch_all returns a tuple of (results, execution_time)"""
+        results = [{"id": 1, "lemma": "test"}]
+        mock_db_connection[0].cursor.fetchall.return_value = results
+
+        with patch("db_mysql._fetch_all", return_value=(results, 0.5)):
+            result, exec_time = fetch_all("SELECT * FROM test")
```

Ruff (0.15.12)

[error] Undefined name `fetch_all`

(F821)